src/main/java/wolox/training/controllers/UserController.java

```diff
@@ -22,16 +23,31 @@
 @RequestMapping("/api/users")
 public class UserController {
 
+    private final IAuthentication authentication;
     private final UserRepository userRepository;
     private final BookRepository bookRepository;
 
-    public UserController(UserRepository userRepository, BookRepository bookRepository) {
+    public UserController(IAuthentication authentication, UserRepository userRepository,
+        BookRepository bookRepository) {
 
+        this.authentication = authentication;
         this.userRepository = userRepository;
         this.bookRepository = bookRepository;
 
     }
 
+    /**
+     * Method that gets authenticated user
+     *
+     * @return username (String)
+     */
+    @GetMapping("/principal")
```

Comment:
the most common naming strategies for the logged user info are "/me", "/user" 🙂 No need to change it, just a comment 😅